anchore_engine/analyzers/hints.py

```diff
@@ -0,0 +1,407 @@
+import json
+
+from anchore_engine.utils import ensure_str
+
+
+class HintsError(Exception):
+    pass
+
+
+class Hint:
+    def __init__(self, pkg, type):
+        self.type = type
+        self.name = ensure_str(pkg.get("name", ""))
+        self.version = ensure_str(pkg.get("version", ""))
+        self.origin = ensure_str(pkg.get("origin", ""))
+
+    def check_required_fields(self):
+        if not self.name or not self.version or not self.type:
+            raise HintsError(
+                "bad hints record, all hints records must supply at least a name, version and type"
+            )
```

Comment:
This is usually a `TypeError`, and you can use `any()` here to check for the values instead of `or not`:

```suggestion
        if not any([self.name, self.version, self.type]):
            raise TypeError(
                "bad hints record, all hints records must supply at least a name, version and type"
            )
```

Reply:
I added the `any` implementation and it broke the unit test, did you mean `all`?

Reply:
I probably had it backwards - you are closer to the logic, `all` seems like it would do the inverse and cover the unittest. Go for it!